### PR TITLE
feat: Duration plugin, see: https://github.com/iamkun/dayjs/issues/564

### PR DESCRIPTION
**Pull request by lockee14**
Add a duration plugin to solve this issue https://github.com/iamkun/dayjs/issues/564

file added:
-src/plugin/duration/index.js
-test/plugin/duration.test.js
-types/plugin/duration.d.ts

all tests pass without error

durationISO8601

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#564: Feature request: Add Duration Plugin](https://issuehunt.io/repos/128907699/issues/564)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: src/plugin/duration/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+export default (o, c) => {
+  /**
+    * convert a number to a ISO 8601 duration
+    * @param  {number} value a number
+    * @param  {string} unit unit of time of the number
+    * @return {object} duration object
+  */
+  c.prototype.duration = function (value, unit) {
+    const TIME_CONSTANT = {
+      seconde: { u: 60, s: 's', next: 'minute' },
```
Comment (oozusta) [inline src/plugin/duration/index.js]:
```suggestion
      second: { u: 60, s: 's', next: 'minute' },
```

File: test/plugin/duration.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+it('should return an empty object', () => {
+  expect(dayjs().duration()).toEqual({})
+})
+
+it('should return an empty object', () => {
+  expect(dayjs().duration(10, 'badunit')).toEqual({})
+})
+
+it('should return the duration in ISO 8601 format', () => {
+  expect(dayjs().duration(90, 'seconde').ISO).toBe('P1M30S')
```
Comment (oozusta) [inline test/plugin/duration.test.js]:
```suggestion
  expect(dayjs().duration(90, 'second').ISO).toBe('P1M30S')
```

**Comment by mishaaq** [comment]
Is this going to be merged?

**Comment by iamkun** [comment]
#858

**Pull request closed without merging**